src/commcare_cloud/commands/migrations/copy_files.py

```diff
@@ -303,6 +310,3 @@ def _set_auth_key(source_context, source_host, source_user,
     state = 'absent' if remove else 'present'
     args = "user={} state={} key={{{{ lookup('file', '{}') }}}}".format(source_user, state, key_path)
     run_ansible_module(source_context, source_host, 'authorized_key', args)
-
-    if remove:
```

Comment:
Oh, I think I understand. If this also checked `os.path.exists(key_file)` I think it would work. Seems like the AI hallucinated the race condition.